test/e2e_node/mount_rro_linux_test.go

```diff
@@ -0,0 +1,434 @@
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+		},
+	}
+}
+
+func podForRROUnsupported(name, ns string) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			InitContainers: []v1.Container{
+				{
+					Image:   busyboxImage,
+					Name:    "mount",
+					Command: []string{"sh", "-euxc", "mkdir -p /mnt/tmpfs && mount -t tmpfs none /mnt/tmpfs"},
+					SecurityContext: &v1.SecurityContext{
+						Privileged: ptr.To(true),
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:             "mnt",
+							MountPath:        "/mnt",
+							MountPropagation: ptr.To(v1.MountPropagationBidirectional),
+						},
+					},
+				},
+				{
+					Image: busyboxImage,
+					Name:  "test",
+					Command: []string{"sh", "-euxc", `
+for f in rro-if-possible rro-disabled ro rw; do touch /mnt-$f/tmpfs/foo; done
+`},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:              "mnt",
+							MountPath:         "/mnt-rro-if-possible",
+							ReadOnly:          true,
+							RecursiveReadOnly: ptr.To(v1.RecursiveReadOnlyIfPossible),
+						},
+						{
+							Name:              "mnt",
+							MountPath:         "/mnt-rro-disabled",
+							ReadOnly:          true,
+							RecursiveReadOnly: ptr.To(v1.RecursiveReadOnlyDisabled), // explicit
+						},
+						{
+							Name:      "mnt",
+							MountPath: "/mnt-ro",
+							ReadOnly:  true,
+						},
+						{
+							Name:      "mnt",
+							MountPath: "/mnt-rw",
+						},
+					},
+				},
+				{
+					Image:   busyboxImage,
+					Name:    "unmount",
+					Command: []string{"umount", "/mnt/tmpfs"},
+					SecurityContext: &v1.SecurityContext{
+						Privileged: ptr.To(true),
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:             "mnt",
+							MountPath:        "/mnt",
+							MountPropagation: ptr.To(v1.MountPropagationBidirectional),
+						},
+					},
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Image:   busyboxImage,
+					Name:    "completion",
+					Command: []string{"echo", "OK"},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "mnt",
+					VolumeSource: v1.VolumeSource{
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
+				},
+			},
+		},
+	}
+}
+
+func supportsRRO(ctx context.Context, f *framework.Framework) bool {
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	framework.ExpectNoError(err)
+	// Assuming that there is only one node, because this is a node e2e test.
+	gomega.Expect(nodeList.Items).To(gomega.HaveLen(1))
+	node := nodeList.Items[0]
+	for _, f := range node.Status.RuntimeClasses {
+		if f.Name == "" && f.Features != nil && *f.Features.RecursiveReadOnlyMounts {
+			return true
+		}
+	}
+	return false
+}
+
+var _ = SIGDescribe("Mount recursive read-only [LinuxOnly]", framework.WithSerial(), nodefeature.RecursiveReadOnlyMounts, func() {
```

Comment:
nit: i usually prefer calling functions to be above called functions so the file can be read top-down. If you need to address more notes, would you mind reordering?

Reply:
fixed